packages/git/src/browser/git-widget.tsx

```diff
@@ -85,28 +86,29 @@ export class GitWidget extends GitDiffWidget implements StatefulWidget {
 
     @postConstruct()
     protected init() {
-        this.toDispose.push(this.repositoryProvider.onDidChangeRepository(repository =>
```

Comment:
Why should it go in `initialize` method? Does it depend on `repository`?

Reply:
`initialize` is just a function called by `init`, not the special one from `FrontendApplicationContribution`.

the state of the GitWidget is dependent on the value of `RepositoryProvider.selectedRepository` which can be changed by the users when theia is running. Every time `selectedRepository` changes, we need to refresh `GitWidget` to display the correct data.

The reason that i am moving this line into `initialize` function is that `toDispose.dispose()` from `initialize` disposes what this line does. and with the code before my change, the state of GitWidget would be out of sync with the rest of theia unless the user manually refreshes the widget.